lib/nrf_modem_lib/Kconfig.modemlib

```diff
@@ -31,56 +31,52 @@ config NRF_MODEM_LIB_TRACE_ENABLED
 	  Trace data is output on the chosen trace backend.
 	  To stop traces or to change trace level at runtime, use the AT command %XMODEMTRACE or the nrf_modem_lib_trace API.
 
-if NRF_MODEM_LIB_TRACE_ENABLED
+if NRF_MODEM_LIB_TRACE
 
-choice NRF_MODEM_LIB_TRACE
+config NRF_MODEM_LIB_TRACE_LEVEL_OVERRIDE
+	bool "Set default trace level upon start"
+	default y
+
+if NRF_MODEM_LIB_TRACE_LEVEL_OVERRIDE
+
+choice
```

Comment:
Looks like the choice needs a name to compile. Obvious name is `NRF_MODEM_LIB_TRACE_LEVEL`. However that is taken by the integer config below. Maybe rename the integer config to `NRF_MODEM_LIB_TRACE_LEVEL_INT`? This is never set by the application, but depends solely on the choice of the trace level. Then the choice can be named `NRF_MODEM_LIB_TRACE_LEVEL`.